benchmark/src/Extension/EvergreenReport.php

```diff
@@ -0,0 +1,115 @@
+<?php
+
+namespace MongoDB\Benchmark\Extension;
+
+use PhpBench\Compat\SymfonyOptionsResolverCompat;
+use PhpBench\Model\Suite;
+use PhpBench\Model\SuiteCollection;
+use PhpBench\Path\Path;
+use PhpBench\Registry\Config;
+use PhpBench\Report\GeneratorInterface;
+use PhpBench\Report\Model\Reports;
+use RuntimeException;
+use Symfony\Component\OptionsResolver\OptionsResolver;
+
+use function dirname;
+use function file_exists;
+use function file_put_contents;
+use function json_encode;
+use function mkdir;
+use function sprintf;
+
+use const JSON_PRETTY_PRINT;
+
+class EvergreenReport implements GeneratorInterface
+{
+    private const PARAM_PATH = 'path';
+
+    public function __construct(private string $cwd)
+    {
+    }
+
+    public function configure(OptionsResolver $options): void
+    {
+        $options->setDefaults([
+            self::PARAM_PATH => '.phpbench/results.json'
+        ]);
+        $options->setAllowedTypes(self::PARAM_PATH, ['string']);
+
+        SymfonyOptionsResolverCompat::setInfos($options, [
+            self::PARAM_PATH => 'Path to output file',
+        ]);
+    }
+
+    public function generate(SuiteCollection $collection, Config $config): Reports
+    {
+        $tests = [];
+
+        /** @var Suite $suite */
+        foreach ($collection as $suite) {
+            foreach ($suite as $benchmark) {
+                foreach ($benchmark as $subject) {
+                    foreach ($subject->getVariants() as $variant) {
+                        $stats = $variant->getStats()->getStats();
+                        $name = sprintf('%s::%s', $benchmark->getName(), $subject->getName());
+                        if ($variant->getParameterSet()->getName()) {
+                            $name .= '#' . $variant->getParameterSet()->getName();
+                        }
+
+                        $tests[] = [
+                            'info' => [
+                                'test_name' => $name,
+                            ],
+                            'created_at' => date(DATE_ATOM),
+                            'completed_at' => date(DATE_ATOM),
```

Comment:
I don't think this information is relevant to us, so I didn't bother trying to extract the exact dates. The documentation also didn't mark this as optional, so I decided to just include the current date instead.

Reply:
Well, `created_at` could take `$_SERVER['REQUEST_TIME']`.